packages/cozy-stack-client/src/getIconURL.preloader.js

```diff
@@ -0,0 +1,100 @@
+const _loaded = {}
+
+/**
+ * Preload Icon and save it into _loaded singleton
+ *
+ * @param  {object}  app - Apps data - io.cozy.apps
+ * @param  {string|undefined} domain - Host to use in the origin (e.g. cozy.tools)
+ * @param  {string} protocol - Url protocol (e.g. http / https)
+ * @returns {Promise<string>}  Source Url of loaded icon
+ * @throws {Error} When cannot fetch or preload icon
+ */
+export const preload = async (app, domain, protocol) => {
+  console.log('inside preload')
+  console.log({ app, domain, protocol })
+  console.log({ _loaded })
+
+  if (!domain) throw new Error('Cannot fetch icon: missing domain')
+
+  const source = _getAppIconURL(app, domain, protocol)
+  console.log('source')
+
+  if (!source) {
+    throw new Error(`Cannot get icon source for app ${app.name}`)
+  }
+
+  console.log('je return une promise')
+  return new Promise((resolve, reject) => {
+    console.log('inside promise')
+    const loader = document.createElement('img')
+    console.log({  loader })
+    loader.onload = () => {
+      console.log('inside on load')
+      _loaded[source] = true
+      resolve(source)
+    }
+    loader.onerror = () => {
+      console.log('inside on error')
+      const loadError = new Error(`Error while preloading ${source}`)
+      console.error(loadError.message)
+      reject(loadError)
+    }
+    loader.src = source
+    console.log('loader end, all set:')
+    console.log({loader})
+
+  })
+}
```

Comment:
not sure that this `preload/ed` stuff is needed. I don't see the point to have it now. Cozy-Stack should send the correct header to put it in cache, no?

Reply:
exact, I'm updating the PR right now.